bsp_diff/caas/device/intel/civ/host/vm-manager/0046-Multi-Android-VM-support.patch

```diff
@@ -0,0 +1,403 @@
++            echo "Error: $file file is missing"
++            exit -1
++        fi
++    done
++
++    cp $CIV_DIR/OVMF.fd $CIV_DIR/$3/
++    cp $CIV_DIR/scripts/rpmb_dev $CIV_DIR/$3/scripts/
++}
++
++function create_vm_dirs() {
++    echo "Dir: $CIV_DIR/$3"
++    mkdir $CIV_DIR/$3
++    $CIV_DIR/scripts/setup_civ_ini.sh -v $1 -p $2 -n $3
++}
++
++
++function create_vms() {
++    create_vm_dirs $1 $2 $3
++    copy_files_for_vm_flashing $1 $2 $3
++}
++
++function setup_files() {
++    echo "Creating setup for VMs"
++    if [ -d $USER_HOME/.intel/.civ ]
++    then
++        SETUP_HOST_DONE=true
++    fi
++    new_dir=$CIV_DIR/$VM_NAME
++    if [ -n "$(ls -d ${new_dir}?)" ]; then
++	dir_arr=$(ls -d ${new_dir}?)
++	directories=( $dir_arr )
++	last_vm=${directories[-1]}
++	final_vm_name="${last_vm##*/}"
++	last_character=${final_vm_name: -1}
++	last_character=$((last_character+1))
++	ADB_PORT=$((5555+(last_character*2)))
++	VSOCK_ID=$((3+last_character))
++	INITIAL_VM_NUM=$last_character
++    fi
++
++    for (( i=$INITIAL_VM_NUM; i<$NUMBER_OF_VM+$INITIAL_VM_NUM; i++ ))
++    do
++	echo "Folder name: $CIV_DIR/${VM_NAME}${i}"
++
++	if [ $SETUP_HOST_DONE = false ]; then
++            ./scripts/setup_host.sh -n ${VM_NAME}${i}
++	    SETUP_HOST_DONE=true
++	else
++	    ADB_PORT=$((ADB_PORT+(i*2)))
++	    create_vms $((i+3)) $ADB_PORT ${VM_NAME}${i}
++	fi
++    done
++
++    echo "Please reboot"
++}
++
++function copy_files_for_vm() {
++    echo "Copying to folder $VM_NAME$1"
++    if [ -f "$CIV_DIR/$VM_NAME$INITIAL_VM_NUM/android.qcow2" ]; then
++        cp $CIV_DIR/$VM_NAME$INITIAL_VM_NUM/android.qcow2 $CIV_DIR/$VM_NAME$1
++    else
++        echo "android.qcow2 missing. Please run setup_host.sh, start_flash_usb.sh and try again"
++        exit
++    fi
++
++    if [ -d "$CIV_DIR/$VM_NAME$INITIAL_VM_NUM/vtpm0" ]; then
++        cp -r $CIV_DIR/$VM_NAME$INITIAL_VM_NUM/vtpm0 $CIV_DIR/$VM_NAME$1
++    else
++        echo "vtpm0 missing. Please run setup_host.sh, start_flash_usb.sh and try again"
++        exit
++    fi
++}
++
++function flash_vms() {
++    echo "Flashing VMs"
++    if [ -f caas-flashfiles-*.zip ]; then
++        ./scripts/start_flash_usb.sh caas-flashfiles-*.zip --display-off -n $VM_NAME$INITIAL_VM_NUM
++    else
++        echo "flashfiles missing. Please download and unzip the package correctly."
++        exit -1
++    fi
++
++    for (( i=$INITIAL_VM_NUM+1; i<$NUMBER_OF_VM+$INITIAL_VM_NUM; i++ ))
++    do
++        copy_files_for_vm $i
++        echo "Flashed ${VM_NAME}${i}"
++    done
++    echo "Done flashing"
++}
++
++function show_help() {
++    printf "Creates specified number of Android guests under folder name Android-CIVx, where x is the guest number.\n"
++    printf "Usage: \n"
++    printf "$(basename "$0") [-c] \n"
++    printf "Options:\n"
++    printf "\t-h  show this help message\n"
++    printf "\t-c  specify number of Android guests to be newly created. Default without specifying -c, is 1.\n"
```

Comment:
Actual output:

> tester@tester-NUC:~$ sudo ./scripts/setup_multi_civ_vm.sh --help
> Creates specified number of Android guests under folder name Android-CIVx, where x is the guest num                                                             ber.
> Usage:
> setup_multi_civ_vm.sh [-c]
> Options:
>         -h  show this help message
>         -c  specify number of Android guests to be newly created. Default without specifying -c, is                                                              1.
> 

Please shorten the lines so that it does not have to wrap around